components/templates/ListTodos/index.tsx

```diff
@@ -0,0 +1,66 @@
+import { useMutation } from 'react-query'
+import React, { useCallback, useContext, useEffect } from 'react'
+
+import { TodoContext } from '@/context/TodoContext'
+import { apiRequest } from '@/helpers/common/lib/axios'
+import { TodoType } from '@/helpers/common/types/todoType'
+import TodoItem from './TodoItem'
+import LoadingText from '@/components/common/LoadingText'
+
+type Todos = {
+  todos: TodoType[]
+}
+
+const ListTodos = () => {
+  const { todos, setTodos } = useContext(TodoContext)
+
+  const todosMutationFn = async (): Promise<TodoType[]> => {
+    const { todos } = await apiRequest<Todos>({
+      method: 'POST',
+      url: '/api/todos/get',
+      data: JSON.stringify({}),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    })
+
+    return todos
+  }
+
+  const { mutateAsync: getTodosMutationHandler, isLoading: isLoadingTodos } =
+    useMutation({
+      mutationKey: 'GET_TODO',
+      mutationFn: todosMutationFn,
+    })
+
+  const fetchTodos = useCallback(async () => {
+    const todos = await getTodosMutationHandler()
+    setTodos([...todos])
+  }, [])
+
+  useEffect(() => {
+    let mounted = true
+
+    if (mounted) {
+      fetchTodos()
+    }
+
+    return () => {
+      mounted = false
+    }
+  }, [])
+
+  if (isLoadingTodos) {
```

Comment:
I think by using react query's loading handler in the useQuery hook, these use effects would be unecessary.

https://tanstack.com/query/v3/docs/react/examples/react/basic